chain/stmgr/forks.go

```diff
@@ -756,3 +819,34 @@ func resetGenesisMsigs(ctx context.Context, sm *StateManager, store adt0.Store,
 
 	return nil
 }
+
+func resetMultisigVesting(ctx context.Context, store adt0.Store, tree *state.StateTree, addr address.Address, startEpoch abi.ChainEpoch, duration abi.ChainEpoch, balance abi.TokenAmount) error {
+	act, err := tree.GetActor(addr)
+	if err != nil {
+		return xerrors.Errorf("getting actor: %w", err)
+	}
+
+	if !builtin.IsMultisigActor(act.Code) {
+		return xerrors.Errorf("actor wasn't msig: %w", err)
+	}
+
+	var msigState multisig0.State
+	if err := store.Get(ctx, act.Head, &msigState); err != nil {
+		return xerrors.Errorf("reading multisig state: %w", err)
+	}
+
+	msigState.StartEpoch = startEpoch
+	msigState.UnlockDuration = duration
```

Comment:
Nit: I think this is ok in practise, but note that blindly setting these can break actor state invariants. The state that these msig actors are in now, with nonzero StartEpoch but zero UnlockDuration is in fact impossible to reach by normal means. It's only possible because the prior upgrade manipulated state directly, into a configuration that the actor code is actually not prepared to handle. The code, for better or worse, expects nonzero `InitialBalance` and `StartEpoch` iff `UnlockDuration!=0`. See the actor constructor and `LockBalance` method.

I *think* that nothing bad will come of it here (except the undesirable lockup that this change reverts), but it's not a good practise. The "impossible" state configurations can have no test coverage in the actors project.

In this PR, the callers all seem to set `(0, 0, 0)`, which is a supported configuration.

For the future, I recommend:
- careful study of the code of actors which state you intend to manipulate directly
- extraction and maintenance of the same invariants enforced by the actor code
- requiring review from an actors maintainer
- [actors team] pushing invariants lower down the call tree so they're easier to include by default